filters/auth/tokeninfo_test.go

```diff
@@ -575,3 +578,126 @@ func TestOAuthTokeninfoAllocs(t *testing.T) {
 		t.Errorf("Expected zero allocations, got %f", allocs)
 	}
 }
+
+func TestOAuthTokeninfoValidate(t *testing.T) {
+	backend := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, _ *http.Request) {
+		w.Write([]byte("OK"))
+	}))
+	defer backend.Close()
+
+	const validAuthHeader = "Bearer foobarbaz"
+
+	var authRequestsTotal atomic.Int32
+	testAuthServer := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		authRequestsTotal.Add(1)
+		if r.Header.Get("Authorization") != validAuthHeader {
+			w.WriteHeader(http.StatusUnauthorized)
+		} else {
+			w.Write([]byte(`{"uid": "foobar", "scope":["foo", "bar"]}`))
+		}
+	}))
+	defer testAuthServer.Close()
+
+	tio := TokeninfoOptions{
+		URL:     testAuthServer.URL,
+		Timeout: testAuthTimeout,
+	}
+
+	for _, tc := range []struct {
+		name               string
+		precedingFilters   string
+		authHeader         string
+		expectStatus       int
+		expectAuthRequests int32
+	}{
+		// reject
+		{
+			name:               "reject missing token",
+			authHeader:         "",
+			expectStatus:       http.StatusUnauthorized,
+			expectAuthRequests: 0,
+		},
+		{
+			name:               "reject invalid token",
+			authHeader:         "Bearer invalid",
+			expectStatus:       http.StatusUnauthorized,
+			expectAuthRequests: 1,
+		},
+		{
+			name:               "reject invalid token type",
+			authHeader:         "Basic foobarbaz",
+			expectStatus:       http.StatusUnauthorized,
+			expectAuthRequests: 0,
+		},
+		{
+			name:               "reject missing token when opt-out is invalid",
+			precedingFilters:   `annotate("oauth.invalid", "invalid opt-out")`,
+			authHeader:         "",
+			expectStatus:       http.StatusUnauthorized,
+			expectAuthRequests: 0,
+		},
+		// allow
```

Comment:
these kind of comments will be outdated to soon to make sense, IMO

Reply:
These were to mark two test case groups as I did not want to add two t.Run, removed.